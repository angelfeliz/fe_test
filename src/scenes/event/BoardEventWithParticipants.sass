
.header-table__container
    width: 100%
    background-color: black
    color: white
    display: flex
    flex-direction: column
    min-height: 10vh
    height: 90px
    padding-top: 5%
    justify-content: flex-end
    background-image: url('../../imagenes/banner_corrugated.png')
    background-repeat: no-repeat
    background-size: cover
    

.header--font-small
    font-size: .6rem
    font-weight: bold

.header-table__title 
    margin-bottom: 3%

.title
    display: flex
    flex-direction: row
    margin-left: 5%
    margin-bottom: 6%

.header-table
    display: flex
    flex-direction: row
    align-items: flex-end

.bib, .age
    width: 14%
    text-align: center

.gender, .time, .score
    width: 16%
    text-align: center


@media only screen and (min-width: 992px)
    .header-table__container
         flex-direction: column
         height: 120px
        
    .header-table__title
        flex-flow: row
        margin-left: 5%
    
    .header-table
        align-items: flex-end


