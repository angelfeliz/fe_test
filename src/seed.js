import face1 from './imagenes/face1.png'
import face2 from './imagenes/face2.png'

export const data =  {
    eventName: 'post apocalyptic highway',
    participants: [
        {
            position: 1,
            fullName: 'Angel Rodriguez',
            photo: face1,
            bib: 1085,
            age: 27,
            gender: 'M',
            time: '17:42',
            score: 41,
            events: [
                {
                    challenge: 'post apocalyptic highway',
                    activities: [
                        {
                          order: 1,
                          name: 'Hill Climb',
                          P: true,
                          R: true,
                          W: true,
                          F: true,
                          skillP: 20,
                          skillR: 40,
                          skillW: 100,
                          skillF: 0,
                          time: '17:90',
                          score: 8
                        },
                        {
                          order: 2,
                          name: 'Run For The Hills',
                          P: true,
                          R: true,
                          W: false,
                          F: false,
                          skillP: 20,
                          skillR: 40,
                          skillW: 100,
                          skillF: 0,
                          time: '17:90',
                          score: 8
                        },
                        {
                          order: 3,
                          name: 'Coded Doors',
                          P: true,
                          R: false,
                          W: true,
                          F: false,
                          skillP: 20,
                          skillR: 40,
                          skillW: 100,
                          skillF: 0,
                          time: '17:90',
                          score: 8
                        }
                    ]
                },
                {
                    challenge: 'Costa Ricas Canves',
                    activities: [
                        {
                          order: 1,
                          name: 'Hill Climb',
                          P: true,
                          R: true,
                          W: false,
                          F: true,
                          skillP: 20,
                          skillR: 40,
                          skillW: 100,
                          skillF: 0,
                          time: '17:90',
                          score: 8
                        },
                        {
                          order: 2,
                          name: 'Run For The Hills',
                          P: true,
                          R: false,
                          W: false,
                          F: true,
                          skillP: 20,
                          skillR: 40,
                          skillW: 100,
                          skillF: 0,
                          time: '17:90',
                          score: 8
                        }
                    ]
                }
            ]
        },
        {
            position: 2,
            fullName: 'Patrick Johnson',
            photo: face2,
            bib: 1085,
            age: 27,
            gender: 'M',
            time: '17:42',
            score: 41,
            events:[]
        }
    ]
}