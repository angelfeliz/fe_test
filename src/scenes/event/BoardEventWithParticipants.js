import React from 'react';
import {data} from '../../seed';
import './BoardEventWithParticipants.sass';
import ParticipantRow from '../../components/Participant/ParticipantRow';

class BoardEventWithParticipants extends React.PureComponent {
    constructor(props){
        super(props);
        this.state = {
            rowIndexShowDetail: undefined
        };
       
    }
    showDetail = (index) => {
        
        if(index !== this.state.rowIndexShowDetail) {
            this.setState({rowIndexShowDetail: index});
            console.log(index);
        }
        else{
            this.setState({rowIndexShowDetail: undefined});
        }
    }

    render() {
        let headers =
        [
            {name:'BIB', class:'bib'},
            {name:'AGE', class:'age'},
            {name:'GENDER', class:'gender'},
            {name:'TIME', class:'time'},
            {name:'SCORE', class:'score'}
        ];

        return(
            <React.Fragment>
                <div className='header-table__container'>
                    <div className='header-table__title font--big flex md-fl-90'>{data.eventName}</div>
                    <div className='header-table flex'>
                        <div className="flex xs-fl-30 sm-fl-20 md-fl-90"></div>
                        {
                            headers.map((item, index) => {
                                return(
                                    <div key={index} className={`header--font-small ${item.class}`}>{item.name}</div>
                                );
                            })
                        }
                    </div>
                </div>

                {
                    data.participants.map((item, index) => {
                                let openDetail = index === this.state.rowIndexShowDetail;
                                console.log(openDetail);
                                return (
                                    <ParticipantRow
                                        key={index}
                                        {...item}
                                        index={index}
                                        showDetail={this.showDetail}
                                        openDetail={openDetail}
                                    />
                                );
                            })
                }
            </React.Fragment>
        )
    }
}

export default BoardEventWithParticipants;