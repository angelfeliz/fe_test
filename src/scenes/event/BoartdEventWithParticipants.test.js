import 'jest-dom/extend-expect';
import 'react-testing-library/cleanup-after-each';

import React from 'react'; 
import { render } from 'react-testing-library';
import { data } from '../../seed';
import BoardEventWithParticipants from './BoardEventWithParticipants';

test('Snapshot of Board', () => {
    const { container } = render(<BoardEventWithParticipants/>);
    expect(container).toMatchSnapshot();
});