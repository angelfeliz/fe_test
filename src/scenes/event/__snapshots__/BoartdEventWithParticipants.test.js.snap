// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Board 1`] = `
<div>
  <div
    class="header-table__container"
  >
    <div
      class="header-table__title font--big flex md-fl-90"
    >
      post apocalyptic highway
    </div>
    <div
      class="header-table flex"
    >
      <div
        class="flex xs-fl-30 sm-fl-20 md-fl-90"
      />
      <div
        class="header--font-small bib"
      >
        BIB
      </div>
      <div
        class="header--font-small age"
      >
        AGE
      </div>
      <div
        class="header--font-small gender"
      >
        GENDER
      </div>
      <div
        class="header--font-small time"
      >
        TIME
      </div>
      <div
        class="header--font-small score"
      >
        SCORE
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="row__section-1"
      data-testid="rowitem"
    >
      <div
        class="row--number-position flex xs-fl-10 font--small"
      >
        <span>
          1
        </span>
      </div>
      <div
        class="row__inner-row flex"
      >
        <div
          class="flex xs-fl-30 sm-fl-20 md-fl-20 lg-fl-10 row__container__photo"
        >
          <img
            alt="face"
            class="row__photo"
            src="face1.png"
          />
        </div>
        <div
          class="row__inner-row flex xs-fl-70 sm-fl-90 md-fl-90 position_container"
        >
          <div
            class="row__name flex  sm-fl-70 md-fl-40 lg-fl-90 font--normal font--bold"
          >
            <span>
              Angel Rodriguez
            </span>
          </div>
          <div
            class="row__inner-row  row--flex-end flex row__inner--sm-hg"
          >
            <div
              class="row__item font--small row-score--item-bib"
            >
              <span>
                1085
              </span>
            </div>
            <div
              class="row__item font--small row-score--item-age"
            >
              <span>
                27
              </span>
            </div>
            <div
              class="row__item row-score--item-gender"
            >
              <span>
                M
              </span>
            </div>
            <div
              class="row__item font--normal row-score--item-time"
            >
              <span>
                17:42
              </span>
            </div>
            <div
              class="row__item font--normal row--bg-blue row-score--item-score"
            >
              <span>
                41
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row__section-2"
    />
  </div>
  <div
    class="row"
  >
    <div
      class="row__section-1"
      data-testid="rowitem"
    >
      <div
        class="row--number-position flex xs-fl-10 font--small"
      >
        <span>
          2
        </span>
      </div>
      <div
        class="row__inner-row flex"
      >
        <div
          class="flex xs-fl-30 sm-fl-20 md-fl-20 lg-fl-10 row__container__photo"
        >
          <img
            alt="face"
            class="row__photo"
            src="face2.png"
          />
        </div>
        <div
          class="row__inner-row flex xs-fl-70 sm-fl-90 md-fl-90 position_container"
        >
          <div
            class="row__name flex  sm-fl-70 md-fl-40 lg-fl-90 font--normal font--bold"
          >
            <span>
              Patrick Johnson
            </span>
          </div>
          <div
            class="row__inner-row  row--flex-end flex row__inner--sm-hg"
          >
            <div
              class="row__item font--small row-score--item-bib"
            >
              <span>
                1085
              </span>
            </div>
            <div
              class="row__item font--small row-score--item-age"
            >
              <span>
                27
              </span>
            </div>
            <div
              class="row__item row-score--item-gender"
            >
              <span>
                M
              </span>
            </div>
            <div
              class="row__item font--normal row-score--item-time"
            >
              <span>
                17:42
              </span>
            </div>
            <div
              class="row__item font--normal row--bg-blue row-score--item-score"
            >
              <span>
                41
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row__section-2"
    />
  </div>
  <div
    class="row"
  >
    <div
      class="row__section-1"
      data-testid="rowitem"
    >
      <div
        class="row--number-position flex xs-fl-10 font--small"
      >
        <span>
          3
        </span>
      </div>
      <div
        class="row__inner-row flex"
      >
        <div
          class="flex xs-fl-30 sm-fl-20 md-fl-20 lg-fl-10 row__container__photo"
        >
          <img
            alt="face"
            class="row__photo"
            src="face3.png"
          />
        </div>
        <div
          class="row__inner-row flex xs-fl-70 sm-fl-90 md-fl-90 position_container"
        >
          <div
            class="row__name flex  sm-fl-70 md-fl-40 lg-fl-90 font--normal font--bold"
          >
            <span>
              Michael Monks
            </span>
          </div>
          <div
            class="row__inner-row  row--flex-end flex row__inner--sm-hg"
          >
            <div
              class="row__item font--small row-score--item-bib"
            >
              <span>
                1085
              </span>
            </div>
            <div
              class="row__item font--small row-score--item-age"
            >
              <span>
                27
              </span>
            </div>
            <div
              class="row__item row-score--item-gender"
            >
              <span>
                M
              </span>
            </div>
            <div
              class="row__item font--normal row-score--item-time"
            >
              <span>
                17:42
              </span>
            </div>
            <div
              class="row__item font--normal row--bg-blue row-score--item-score"
            >
              <span>
                41
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row__section-2"
    />
  </div>
  <div
    class="row"
  >
    <div
      class="row__section-1"
      data-testid="rowitem"
    >
      <div
        class="row--number-position flex xs-fl-10 font--small"
      >
        <span>
          4
        </span>
      </div>
      <div
        class="row__inner-row flex"
      >
        <div
          class="flex xs-fl-30 sm-fl-20 md-fl-20 lg-fl-10 row__container__photo"
        >
          <img
            alt="face"
            class="row__photo"
            src="face4.png"
          />
        </div>
        <div
          class="row__inner-row flex xs-fl-70 sm-fl-90 md-fl-90 position_container"
        >
          <div
            class="row__name flex  sm-fl-70 md-fl-40 lg-fl-90 font--normal font--bold"
          >
            <span>
              Hakim Callender
            </span>
          </div>
          <div
            class="row__inner-row  row--flex-end flex row__inner--sm-hg"
          >
            <div
              class="row__item font--small row-score--item-bib"
            >
              <span>
                1085
              </span>
            </div>
            <div
              class="row__item font--small row-score--item-age"
            >
              <span>
                27
              </span>
            </div>
            <div
              class="row__item row-score--item-gender"
            >
              <span>
                M
              </span>
            </div>
            <div
              class="row__item font--normal row-score--item-time"
            >
              <span>
                17:42
              </span>
            </div>
            <div
              class="row__item font--normal row--bg-blue row-score--item-score"
            >
              <span>
                41
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row__section-2"
    />
  </div>
  <div
    class="row"
  >
    <div
      class="row__section-1"
      data-testid="rowitem"
    >
      <div
        class="row--number-position flex xs-fl-10 font--small"
      >
        <span>
          5
        </span>
      </div>
      <div
        class="row__inner-row flex"
      >
        <div
          class="flex xs-fl-30 sm-fl-20 md-fl-20 lg-fl-10 row__container__photo"
        >
          <img
            alt="face"
            class="row__photo"
            src="face5.png"
          />
        </div>
        <div
          class="row__inner-row flex xs-fl-70 sm-fl-90 md-fl-90 position_container"
        >
          <div
            class="row__name flex  sm-fl-70 md-fl-40 lg-fl-90 font--normal font--bold"
          >
            <span>
              Everton Lawrence
            </span>
          </div>
          <div
            class="row__inner-row  row--flex-end flex row__inner--sm-hg"
          >
            <div
              class="row__item font--small row-score--item-bib"
            >
              <span>
                1085
              </span>
            </div>
            <div
              class="row__item font--small row-score--item-age"
            >
              <span>
                27
              </span>
            </div>
            <div
              class="row__item row-score--item-gender"
            >
              <span>
                M
              </span>
            </div>
            <div
              class="row__item font--normal row-score--item-time"
            >
              <span>
                17:42
              </span>
            </div>
            <div
              class="row__item font--normal row--bg-blue row-score--item-score"
            >
              <span>
                41
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row__section-2"
    />
  </div>
  <div
    class="row"
  >
    <div
      class="row__section-1"
      data-testid="rowitem"
    >
      <div
        class="row--number-position flex xs-fl-10 font--small"
      >
        <span>
          6
        </span>
      </div>
      <div
        class="row__inner-row flex"
      >
        <div
          class="flex xs-fl-30 sm-fl-20 md-fl-20 lg-fl-10 row__container__photo"
        >
          <img
            alt="face"
            class="row__photo"
            src="face6.png"
          />
        </div>
        <div
          class="row__inner-row flex xs-fl-70 sm-fl-90 md-fl-90 position_container"
        >
          <div
            class="row__name flex  sm-fl-70 md-fl-40 lg-fl-90 font--normal font--bold"
          >
            <span>
              Patricia Sothers
            </span>
          </div>
          <div
            class="row__inner-row  row--flex-end flex row__inner--sm-hg"
          >
            <div
              class="row__item font--small row-score--item-bib"
            >
              <span>
                1085
              </span>
            </div>
            <div
              class="row__item font--small row-score--item-age"
            >
              <span>
                27
              </span>
            </div>
            <div
              class="row__item row-score--item-gender"
            >
              <span>
                M
              </span>
            </div>
            <div
              class="row__item font--normal row-score--item-time"
            >
              <span>
                17:42
              </span>
            </div>
            <div
              class="row__item font--normal row--bg-blue row-score--item-score"
            >
              <span>
                41
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row__section-2"
    />
  </div>
  <div
    class="row"
  >
    <div
      class="row__section-1"
      data-testid="rowitem"
    >
      <div
        class="row--number-position flex xs-fl-10 font--small"
      >
        <span>
          7
        </span>
      </div>
      <div
        class="row__inner-row flex"
      >
        <div
          class="flex xs-fl-30 sm-fl-20 md-fl-20 lg-fl-10 row__container__photo"
        >
          <img
            alt="face"
            class="row__photo"
            src="face7.png"
          />
        </div>
        <div
          class="row__inner-row flex xs-fl-70 sm-fl-90 md-fl-90 position_container"
        >
          <div
            class="row__name flex  sm-fl-70 md-fl-40 lg-fl-90 font--normal font--bold"
          >
            <span>
              Pat Calhoun
            </span>
          </div>
          <div
            class="row__inner-row  row--flex-end flex row__inner--sm-hg"
          >
            <div
              class="row__item font--small row-score--item-bib"
            >
              <span>
                1085
              </span>
            </div>
            <div
              class="row__item font--small row-score--item-age"
            >
              <span>
                27
              </span>
            </div>
            <div
              class="row__item row-score--item-gender"
            >
              <span>
                M
              </span>
            </div>
            <div
              class="row__item font--normal row-score--item-time"
            >
              <span>
                17:42
              </span>
            </div>
            <div
              class="row__item font--normal row--bg-blue row-score--item-score"
            >
              <span>
                41
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row__section-2"
    />
  </div>
</div>
`;
